modules/javafx.controls/src/main/java/javafx/scene/control/skin/ScrollPaneSkin.java

```diff
@@ -891,13 +891,8 @@ else if (newVVal < vsb.getMin()) {
             */
             if (vsb.getVisibleAmount() < vsb.getMax()) {
                 double vRange = getSkinnable().getVmax()-getSkinnable().getVmin();
-                double vPixelValue;
-                if (nodeHeight > 0.0) {
-                    vPixelValue = vRange / nodeHeight;
-                }
-                else {
-                    vPixelValue = 0.0;
-                }
+                double vPixelValue = vRange / (nodeHeight - contentHeight);
+                vPixelValue = Double.isFinite(vPixelValue) ? vPixelValue : 0.0;
```

Comment:
I liked the previous logic better and would have just changed the comparison to check `(nodeHeight - contentHeight) > 0.0` (possibly saving the adjusted width or height in a local various to avoid doing it twice). As it is, this is replacing an explicit check on the source values with an out of range check on the result, which seems less intuitive. Also, the previous code did a `> 0` test and the new code effectively does a `!= 0` test.

Reply:
I changed it to check the denominator values instead.